codegen/src/main/resources/templates/query/ModelStaxUnmarshaller.ftl

```diff
@@ -141,9 +151,8 @@ public class ${shape.shapeName}Unmarshaller implements Unmarshaller<${shape.shap
             } else if (xmlEvent.isEndElement()) {
                 if (context.getCurrentDepth() < originalDepth) {
 <#list shape.members as memberModel>
-    <#if memberModel.map && (!memberModel.http.location?? || memberModel.http.location == "headers")>
-                    ${shape.variable.variableName}.${memberModel.fluentSetterMethodName}(${memberModel.variable.variableName});
-
+    <#if (memberModel.map || memberModel.list) && (!memberModel.http.location?? || memberModel.http.location == "headers")>
```

Comment:
Why add list to this? This conditional is crazy complicated, can you add a comment explaining the intent?